### ISSUE TITLE
There is no proper tutorial

### ISSUE THREAD

**Issue by enesMesut** [indygreg/PyOxidizer#484]
Hi i want to use PyOxidizer  for creating standalone executable file but i cant understand your documentation. Its so complex and there is no tutorial that clearly explain process from beginning to end. The tutorial in documentation is too basic and does not meet needs.

**Comment by takov751** [comment]
Greetings I perfectly understand your pain there. I am not a contributor or anything, but done a quick example repository, which might give you some ideas. It's just an example not a tutorial. No explanatory in there just a Dockerfile, python script and configuration file

This repo builds straight into a docker based on scratch only include the built binary. 

https://github.com/takov751/Pyoxidizer_example

Hope you find it useful

**Comment by takov751** [comment]
From here on I might write a basic tutorial , but this tool has a really massive set of options, hence tutorial cannot be written that easily for any occasion

**Comment by enesMesut** [comment]
> Greetings I perfectly understand your pain there. I am not a contributor or anything, but done a quick example repository, which might give you some ideas. It's just an example not a tutorial. No explanatory in there just a Dockerfile, python script and configuration file
> 
> This repo builds straight into a docker based on scratch only include the built binary.
> 
> https://github.com/takov751/Pyoxidizer_example
> 
> Hope you find it useful The built file can be copied out from the built image, however with the modification of the Dockerfile a somewhat useful building environment can be done specifically to your use case @indygreg Do you think this Dockerfile could meet your needs with modification to mould it into a building environment for pyoxidizer ?

Hi, thanks for your effort. I reviewed your bzl file and tried to manipulate in order to fit my project but i could not create a executable, it gave a nonsense error. I have a flask application which imports classes from different py files and i'm trying to make a standalone executable. I give up on pyoxidizer i realized that i wont figure out how to create executable with current state of documentation.

**Comment by enesMesut** [comment]
> 



> From here on I might write a basic tutorial , but this tool has a really massive set of options, hence tutorial cannot be written that easily for any occasion

its too bad to create such massive tool without writing a proper documentation to inform user about how to use it.

**Comment by takov751** [comment]
Well nevertheless the project is really big for such a small contributer base. In case you have a flask application, i would recommend maybe put it in a docker. Otherwise what i have seen in the documentation pyoxidizer capable of scanning and digesting whole venv, which would make it easier for your. However the output binary would be somewhere around ~80-200MB ish depending on your usecase. I would still recommend you do go for a docker solution

**Comment by enesMesut** [comment]
> Well nevertheless the project is really big for such a small contributer base. In case you have a flask application, i would recommend maybe put it in a docker. Otherwise what i have seen in the documentation pyoxidizer capable of scanning and digesting whole venv, which would make it easier for your. However the output binary would be somewhere around ~80-200MB ish depending on your usecase. I would still recommend you do go for a docker solution

It's already dockerized but i dont wanna ship my applications as py files to customer environment directly without creating an executable from it.

**Comment by RonDingDing** [comment]
> Hi i want to use PyOxidizer for creating standalone executable file but i cant understand your documentation. Its so complex and there is no tutorial that clearly explain process from beginning to end. The tutorial in documentation is too basic and does not meet needs.

Agree! I just want to pack a py package with inborn python installed and several pip packages included, and distribute it in Linux, Windows and MacOS, but I don't know how to choose a version, I don't know how to write that bzl language and I don't know how to pack. Getting started just get me to printing, but nobody want to print anything, they want a compiled package.
